Removed the redundant data

rsarwad opened the pull request:
Removed the redundant data from structures, ue_mm_context_t and emm_context_t

(cherry picked from commit bcdc5117f2bc3d1bfab6f6748417310b303c3a2d)

rsarwad committed:
Removed the redundant data structures in ue_mm_context_t and emm_context_t

(cherry picked from commit bcdc5117f2bc3d1bfab6f6748417310b303c3a2d)

ssanadhya commented on lte/gateway/c/oai/include/mme_app_ue_context.h:
```diff
@@ -341,11 +341,6 @@ typedef struct ue_mm_context_s {
   /* Basic identifier for ue. IMSI is encoded on maximum of 15 digits of 4 bits,
```
Remove this comment as well

ssanadhya commented on lte/gateway/c/oai/include/mme_app_ue_context.h:
```diff
@@ -341,11 +341,6 @@ typedef struct ue_mm_context_s {
-  uint8_t member_present_mask; // bitmask, see significance of bits below
-#define UE_CTXT_MEMBER_IMEI_SV (1 << 0)
-  imeisv_t imeisv; // IMEISV received in Security mode complete
   // int                    len;
```
Remove this commented line

rsarwad committed:
Resolving merge conflicts

rsarwad committed:
Merge remote-tracking branch 'upstream/master' into rsarwad_removed_redundant_data

rsarwad committed:
Removal of redundant data between UE and emm context
Type: Refactor
Area of relevance: mme_app and nas
Summary:
As part of merging mme_app and nas task, removed redundant data between UE and emm context.
Test plan:
Executed existing sanity test cases

rsarwad committed:
Merge remote-tracking branch 'upstream/master' into rsarwad_removed_redundant_data

rsarwad committed:
Merge remote-tracking branch 'upstream/master' into rsarwad_removed_redundant_data

ssanadhya commented on lte/gateway/c/oai/tasks/nas/emm/NasTransportHdl.c:
```diff
@@ -116,6 +116,8 @@ int emm_proc_uplink_nas_transport(mme_ue_s1ap_id_t ue_id, bstring nas_msg_pP)
   ue_ctx_p = emm_context_get(&_emm_data, ue_id);
```
Can we rename this to emm_ctxt_p as it is a pointer to the emm_context?

ssanadhya commented on lte/gateway/c/oai/include/mme_app_ue_context.h:
```diff
@@ -369,8 +360,6 @@ typedef struct ue_mm_context_s {
 
   /* TODO: Add TAI list */
```
This TODO is not needed any more.

ssanadhya replied:
The commented code above can also be removed as part of this change.

rsarwad replied:
Removed in PR, 766

ssanadhya commented on lte/gateway/c/oai/include/mme_app_ue_context.h:
```diff
@@ -369,8 +360,6 @@ typedef struct ue_mm_context_s {
 
   /* TODO: Add TAI list */
   tai_t serving_cell_tai;
-  tai_list_t tail_list; // Current Tracking area list
```
Let's keep all the TAI related context together, either all in ue_mm_context_t or all in emm_context_t.

rsarwad committed:
Merge remote-tracking branch 'upstream/master' into rsarwad_removed_redundant_data

rsarwad committed:
Merge remote-tracking branch 'upstream/master' into rsarwad_removed_redundant_data

rsarwad committed:
Addressed review comments

rsarwad committed:
Merge remote-tracking branch 'upstream/master' into rsarwad_removed_redundant_data

ssanadhya commented on lte/gateway/c/oai/include/mme_app_ue_context.h:
```diff
@@ -385,68 +363,22 @@ typedef struct ue_mm_context_s {
 
-  // ODB for PS parameters        // Indicates that the status of the operator determined barring for packet oriented services.
-
   // APN-OI Replacement           // Indicates the domain name to replace the APN-OI when constructing the PDN GW
```
Convert this and next few lines into one multi-line comment describing apn_oi_replacement

rsarwad replied:
Done in PR 766

ssanadhya commented on lte/gateway/c/oai/include/mme_app_ue_context.h:
```diff
@@ -385,68 +363,22 @@ typedef struct ue_mm_context_s {
-  // MME IP address for S11       // MME IP address for the S11 interface (used by S-GW)
-  // LOCATED IN mme_config_t.ipv4.s11
-
   // MME TEID for S11             // MME Tunnel Endpoint Identifier for S11 interface.
```
Combine this and next line into one multi-line comment describing mme_teid_s11

rsarwad replied:
Done in PR, 766

ssanadhya commented on lte/gateway/c/oai/tasks/mme_app/mme_app_state_converter.cpp:
```diff
@@ -236,14 +236,8 @@ void MmeNasStateConverter::ue_context_to_proto(
-    ue_ctxt_proto->mutable_imeisv(),
-    &ue_ctxt->imeisv.u.value,
-    IMEISV_BCD8_SIZE);
+  ue_ctxt_proto->set_imsi(ue_ctxt->emm_context._imsi64);
```
The UE context proto message does not have any member field imsi or imsi_len any more. So lines 239 and 240 are not needed.

rsarwad replied:
Addressed in PR 766

ssanadhya commented on lte/gateway/c/oai/tasks/mme_app/mme_app_state_converter.cpp:
```diff
@@ -284,13 +267,8 @@ void MmeNasStateConverter::proto_to_ue_mm_context(
-  strcpy(
-    (char*) state_ue_mm_context->imeisv.u.value,
-    ue_context_proto->imeisv().c_str());
+  state_ue_mm_context->emm_context._imsi64 = ue_context_proto->imsi();
```
The state_ue_mm_context->emm_context struct is updated via the emm_context_to_proto function call. So lines 270 and 271 are not needed.

rsarwad replied:
Updated in PR 766

rsarwad commented:
The changes are available as part of PR, 766
